File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/annotation/NamedQueries.java
package com.coupang.springframework.data.requery.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * NamedQueries
 *
 * @author debop@coupang.com
 * @since 18. 6. 12
 */
@Target({ ElementType.TYPE })
@Retention(RetentionPolicy.RUNTIME)
public @interface NamedQueries {

    NamedQuery[] value();
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/annotation/NamedQueries.kt
package org.springframework.data.requery.annotation

/**
 * NamedQueries
 *
 * @author debop@coupang.com
 * @since 18. 7. 2
 */
@Target(AnnotationTarget.TYPE, AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
annotation class NamedQueries(

    val value: Array<NamedQuery>

)